lib/routes/translate_route.dart

```diff
@@ -0,0 +1,1119 @@
+      }
+
+      if (translation.isAllEmpty) {
+        return APIResponse.fieldEmpty("translation");
+      }
+
+      if (description != null && description.isAllEmpty) {
+        return APIResponse.fieldEmpty("description");
+      }
+
+      final Glossary glossary = Glossary(
+        uuid: Uuid().v4(),
+        term: term,
+        translation: translation,
+        description: description,
+        language: language,
+        modUUID: modUUID,
+      );
+
+      await glossary.insert();
+
+      return APIResponse.success(data: glossary.outputMap());
+    },
+        requiredFields: ["term", "translation", "language"],
+        authConfig: AuthConfig());
+
+    /// List glossaries
+    router.getRoute("/glossary", (req, data) async {
+      Map<String, dynamic> fields = data.fields;
+
+      final Locale? language =
+          fields["language"] != null ? Locale.parse(fields["language"]) : null;
+      final String? modUUID = fields["modUUID"];
+      final String? filter = fields["filter"];
+      int limit =
+          fields["limit"] != null ? int.tryParse(fields["limit"]) ?? 50 : 50;
+      final int skip =
+          fields["skip"] != null ? int.tryParse(fields["skip"]) ?? 0 : 0;
+
+      // Max limit is 50
+      if (limit > 50) {
+        limit = 50;
+      }
+
+      final List<Glossary> glossaries = await Glossary.list(
+          language: language,
+          modUUID: modUUID,
+          filter: filter,
+          limit: limit,
+          skip: skip);
+
+      return APIResponse.success(data: {
+        "glossaries": glossaries.map((e) => e.outputMap()).toList(),
+        "limit": limit,
+        "skip": skip,
+      });
+    }, authConfig: AuthConfig());
+
+    /// Edit glossary
+    router.patchRoute("/glossary/<uuid>", (req, data) async {
+      Map<String, dynamic> fields = data.fields;
+      final String uuid = fields["uuid"]!;
+
+      final String? term = fields["term"];
+      final String? translation = fields["translation"];
+      final String? description = fields["description"];
+      final String? modUUID = fields["modUUID"];
+
+      if (modUUID != null) {
+        MinecraftMod? mod = await MinecraftMod.getByUUID(modUUID);
+        if (mod == null) {
+          return APIResponse.modelNotFound<MinecraftMod>();
+        }
+      }
+
+      if (term != null && term.isAllEmpty) {
+        return APIResponse.fieldEmpty("term");
+      }
+
+      if (translation != null && translation.isAllEmpty) {
+        return APIResponse.fieldEmpty("translation");
+      }
+
+      if (description != null && description.isAllEmpty) {
+        return APIResponse.fieldEmpty("description");
+      }
+
+      Glossary? glossary = await Glossary.getByUUID(uuid);
+      if (glossary == null) {
+        return APIResponse.modelNotFound<Glossary>();
+      }
+
+      glossary = glossary.copyWith(
+        term: term,
+        translation: translation,
+        description: description,
+        modUUID: modUUID,
```

Comment:
If modUUID is null then this glossary will be changed to a global glossary, but if the request JSON does not specifically mark modUUID as null, it will also trigger the same effect, so a handler is needed.